### DISCUSSION TITLE
posts/Go/canal/

### DISCUSSION THREAD

**Discussion by Q1mi** [Q1mi/BlogComments#261] [Announcements]
https://www.liwenzhou.com/posts/Go/canal/ 

 李文周的 Blog 提供 Go 语言中文教程,李文周博客,Go 语言学习之路是我整理的一个免费教程，包含部分视频教程，也是非常适合新手学习Go语言的入门教程。本文介绍了如何使用阿里开源的MySQL数据同步工具canal，包括如何使用docker搭建canal环境和canal server和canal client的使用，最后演示了如何使用go语言读取canal数据。

**Comment by chaomingd** [comment]
官方的示例报错呢

**Comment by Q1mi** [comment]
> @chaomingd
> 官方的示例报错呢

可以看下导入的proto包是否正确，原canal-go使用的是 "github.com/golang/protobuf/proto"
我博客上现在改为我自己 fork版本的 canal-go ，使用的是 "google.golang.org/protobuf/proto"
